### DISCUSSION TITLE
Permission System: Freely set document permissions

### DISCUSSION THREAD

**Discussion by TheMardy** [appwrite/appwrite#2635] [Ideas]
### Background
I’m struggling a bit with the permission system as we can’t freely set permissions.

Say I have an `assignment` document that I as a student (`team:xxx/member`) created and I only want me and my teachers (`team:xxx/owner`) to see. 

As far as I know I have 3 options:
1. Create a function that does the work and make the collection “read only” (sketchy as documents can still be created through the sdk, we would need Collection permission to disable that, together with document permission to complete this use case.)

2. Create a team for every student-teacher pair (not realistic, way too many teams)

3. Create a team role for every student so that each teacher that has this role can see the students assignments. (Many roles of teacher will probably have big impact performance)

### Other Use cases
I can think of a few more cases where a `team member` wants to freely set permissions:

Say I have a registration app for `employees` to submit the hours they have worked. These documents must be visible for `himself` and 'team administration' (`team:xxx/admin`) so that only he and the business owners can see the worked hours.

Say I have a game where a `player` sends private messages to someone in my current game (team). The player creates a `message` and sets the permissions for `user:himself` and `user:receiver`.

### Proposed solution
Make it possible for users to freely set permissions on their documents. 

Maybe missing some important security or performance related details why the permission system works the way it does.

What are your thoughts?

**Comment by stnguyen90** [comment]
Related discussion: https://github.com/appwrite/appwrite/discussions/2689
